api/share/requests/users.py

```diff
@@ -0,0 +1,65 @@
+import re
+from api.share.requests import ValidRequestObject, InvalidRequestObject
+
+
+class GetUsersCountRequest(ValidRequestObject):
+
+    def __init__(self,
+                 registration_bot_chat_id: str,
+                 tg_username: str,
+                 phone: str,
+                 email: str,
+                 first_name: str,
+                 last_name: str):
+        self.registration_bot_chat_id = registration_bot_chat_id
+        self.tg_username = tg_username
+        self.phone = phone
+        self.email = email
+        self.first_name = first_name
+        self.last_name = last_name
+
+    @classmethod
+    def from_dict(cls, input_dict):
+        # Check input parameters.
+        invalid_request = InvalidRequestObject()
```

Comment:
I would love to have a comment explaining why we're instantiating an object for an invalid request here. Maybe make the comment "check input parameters" a bit more verbose. 

As an outside reader, I'm rather expecting something like a "RequestObject" that can become valid or invalid during validation. Not an `InvalidRequestObject`, for which we then check if it has errors. If I misunderstand the logic, the comment would help me.

Reply:
Because InvalidRequestObject has a method to add errors. We have ValidRequestObject and InvalidRequestObject

Reply:
Yes, I understand that. But my "simple" logic says that it's strange that you create an object for an invalid request even before you know whether the request is valid or not (just because that object has a necessary method - but this object is not legacy code, you made it that way). I would expect something like a `Request` object that can have an `.is_valid` attribute, for example. Or a `Request` object that has an `.errors` list (or set) attribute and an `.is_valid()` method that returns `True` if `.errors` is empty and `False` otherwise.

Reply:
We want to return either this class `GetUsersCountRequest` or `InvalidRequestObject` with errors. We try to separete responosibility: `ValidRequestObject` is resposible for correct input for use cases and `InvalidRequestObject` is in resposible for errors.

Reply:
All right, I think I can see your point, just please make sure this approach is explained :) Like this reply you just posted can be well used in a comment.

Reply:
Ok, I got it.